File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
import java.util.Stack;

class Solution {
    int res = 0;
    int temp = 0;
    void helper(TreeNode root, int k){
        if(root == null){
            return;
        }
        helper(root.left, k);
        temp++;
        if(temp == k){
            res = root.val;
        }
        helper(root.right, k);
    }
    public int kthSmallest(TreeNode root, int k) {
        helper(root, k);
        return res;

    }
}

File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.py
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        result = 0
        curr = 1
        def helper(root):
            nonlocal result
            nonlocal curr
            nonlocal k
            if not root:
                return None
            helper(root.left)
            if curr == k:
                result = root.val
            curr = curr + 1
            helper(root.right)
        helper(root)
        return result
            

        